llvm/lib/Target/Mips/MipsBranchExpansion.cpp

```diff
@@ -766,17 +769,40 @@ bool MipsBranchExpansion::handleSlot(Pred Predicate, Safe SafeInSlot) {
         if (std::next(Iit) == FI->end() ||
             std::next(Iit)->getOpcode() != Mips::NOP) {
           Changed = true;
-          TII->insertNop(*(I->getParent()), std::next(I), I->getDebugLoc())
-              ->bundleWithPred();
-          NumInsertedNops++;
+          if (Predicate(*I) ||
```

Comment:
Why this condition is added here?
I guess you need some comment or description in commit msg.

Reply:
I would check again without this condition and add comment.

Reply:
I deleted this condition and the test result of ./build/bin/clang --target=mipsel-freestanding -mcpu=mips3 -fomit-frame-pointer -S 4.c -o 1.s` or without `-O3` all are OK, I would update it. Maybe some reasons I added it now I forget. Thanks!

Reply:
Updated. I added a new function which makes it simpler.

Reply:
@wzssyqa Could you help review, thanks!